synapse/lib/stormtypes.py

```diff
@@ -1972,6 +1972,96 @@ class LibTime(Lib):
                               'desc': 'This yields the current tick count after each time it wakes up.', }}},
+
+        {'name': 'year', 'desc': '''
+        Returns the year part of a time value.
+        ''',
+         'type': {'type': 'function', '_funcname': 'year',
+                  'args': (
+                      {'name': 'tick', 'desc': 'A time value.', 'type': 'time', },
+                  ),
+                  'returns': {'type': 'int', 'desc': 'The year part of the time expression.', }}},
+
+        {'name': 'month', 'desc': '''
+        Returns the month part of a time value.
+        ''',
+         'type': {'type': 'function', '_funcname': 'month',
+                  'args': (
+                      {'name': 'tick', 'desc': 'A time value.', 'type': 'time', },
+                  ),
+                  'returns': {'type': 'int', 'desc': 'The month part of the time expression.', }}},
+
+        {'name': 'day', 'desc': '''
+        Returns the day part of a time value.
+        ''',
+         'type': {'type': 'function', '_funcname': 'day',
+                  'args': (
+                      {'name': 'tick', 'desc': 'A time value.', 'type': 'time', },
+                  ),
+                  'returns': {'type': 'int', 'desc': 'The day part of the time expression.', }}},
+
+        {'name': 'hour', 'desc': '''
+        Returns the hour part of a time value.
+        ''',
+         'type': {'type': 'function', '_funcname': 'hour',
+                  'args': (
+                      {'name': 'tick', 'desc': 'A time value.', 'type': 'time', },
+                  ),
+                  'returns': {'type': 'int', 'desc': 'The hour part of the time expression.', }}},
+
+        {'name': 'minute', 'desc': '''
+        Returns the minute part of a time value.
+        ''',
+         'type': {'type': 'function', '_funcname': 'minute',
+                  'args': (
+                      {'name': 'tick', 'desc': 'A time value.', 'type': 'time', },
+                  ),
+                  'returns': {'type': 'int', 'desc': 'The minute part of the time expression.', }}},
+
+        {'name': 'second', 'desc': '''
+        Returns the second part of a time value.
+        ''',
+         'type': {'type': 'function', '_funcname': 'second',
+                  'args': (
+                      {'name': 'tick', 'desc': 'A time value.', 'type': 'time', },
+                  ),
+                  'returns': {'type': 'int', 'desc': 'The second part of the time expression.', }}},
+
+        {'name': 'dayofweek', 'desc': '''
+        Returns the index (beginning with 0) of the day within the week.
+        ''',
+         'type': {'type': 'function', '_funcname': 'dayofweek',
+                  'args': (
+                      {'name': 'tick', 'desc': 'A time value.', 'type': 'time', },
+                  ),
+                  'returns': {'type': 'int', 'desc': 'The index of the day within week.', }}},
+
+        {'name': 'dayofyear', 'desc': '''
```

Comment:
Its a little counter-intuitive to me that dayofyear and dayofmonth are 0-indexed values (especially since it differs from python datetime: https://docs.python.org/3/library/time.html#time.struct_time).  Any particular reason you went this way?

Reply:
My thought was that all of the `dayofyear` type APIs would be consistently 0 based so they could be used as array index values when iterating arrays or incrementing histogram buckets.